proto/ptf/base_test.py

```diff
@@ -417,7 +441,11 @@ def set_match_key(self, table_entry, t_name, mk):
             mf.add_to(mf_id, table_entry.match)
 
     def set_action(self, action, a_name, params):
-        action.action_id = self.get_action_id(a_name)
+        try:
+            action.action_id = self.get_action_id(a_name)
```

Comment:
it's probably better IMO to check for the return value of `get_action_id` and if it is `None`, fail the test.
Something like:
```
action_id = self.get_action_id(a_name)
if action_id is None:
  self.fail("Failed to get id of action '{}' - perhaps the action name is misspelled?".format(a_name)
action.action_id = action_id
```
otherwise you rely on behavior from the Protobuf library (that raises a `TypeError` when you try to assign `None` to a primitive data type field).
I also got rid of the `%` string formatting, since I'm a big proponent of using the more powerful `format` everywhere :)